### PR TITLE
Add B2Share support #88

### PR DESCRIPTION
**Pull request by micafer**

### CODE REVIEW

**Comment by J535D165** [comment]
It would be very cool to add support for this!

Can you rebase?

**Comment by J535D165** [comment]
It might also be nice to split the filter-files extension from the PR so that we can have a closer look at the design. It looks promising.

**Comment by micafer** [comment]
Code cleaned.

File: datahugger/base.py [hunk 1]
```diff
@@ -284,6 +284,7 @@ def _get_files_recursive(self, url, folder_name=None, base_url=None):
 
         # get the data from URL
         res = requests.get(url)
+        res.raise_for_status()
```
Comment (J535D165) [inline datahugger/base.py:287]:
```suggestion
```

Introduced in #80

File: datahugger/services.py [hunk 2]
```diff
@@ -320,7 +320,7 @@ class MendeleyDataset(DatasetDownloader):
 class OSFDataset(DatasetDownloader):
     """Downloader for OSF repository."""
 
-    REGEXP_ID = r"osf\.io\/(?P<record_id>.*)/"
+    REGEXP_ID = r"osf\.io\/(?P<record_id>[^\/]*)\/{0,1}"
```
Comment (J535D165) [inline datahugger/services.py:323]: Interesting catch. I'm fine with keeping this in this PR.

**Comment by J535D165** [comment]
Apparently, the test fails because I removed the raise_for_status d8a4a4915fead78ef16c8d7842bc0fba638449bc. I'm a bit confused.

File: datahugger/services.py [hunk 3]
```diff
@@ -425,3 +425,25 @@ class SeaNoeDataset(DatasetDownloader):
...
+    META_FILES_JSONPATH = "files[*]"
+
+    # paths to file attributes
+    ATTR_NAME_JSONPATH = "key"
+    ATTR_SIZE_JSONPATH = "size"
+    ATTR_HASH_JSONPATH = "checksum"
+    ATTR_HASH_TYPE_VALUE = "md5"
+
+    def _get_attr_link(self, record, base_url=None):
+        return f"{base_url}/files/{self._params['record_id']}/{record['key']}"
```
Comment (J535D165) [inline datahugger/services.py:449]:
```suggestion

    # paths to file attributes
    ATTR_NAME_JSONPATH = "key"
    ATTR_FILE_LINK_JSONPATH = "ePIC_PID"
    ATTR_SIZE_JSONPATH = "size"
    ATTR_HASH_JSONPATH = "checksum"
    ATTR_HASH_TYPE_VALUE = "md5"
```

This seems to work. I'm clueless about whats going on.

**Comment by J535D165** [comment]
Now it passes. Did they change something to their API since the commit of @micafer 5 hours ago...? 

Nevertheless, thanks a lot, @micafer, for another cool contribution!

**Comment by micafer** [comment]
> Now it passes. Did they change something to their API since the commit of @micafer 5 hours ago...?

It should not have changed.

**Comment by micafer** [comment]
> Apparently, the test fails because I removed the raise_for_status [d8a4a49](https://github.com/J535D165/datahugger/commit/d8a4a4915fead78ef16c8d7842bc0fba638449bc). I'm a bit confused.

It should not affect the tests.
I added to improve the error message shown to the user in case of HTTP error. Without it the HTTP error is hidden and the error message shown is a bit confusing,

**Comment by J535D165** [comment]
It works now! I tested it on the benchmark dataset. Everything looks good.

**Pull request merged**